### PR TITLE
Implement verify UI

### LINKED ISSUE #9: Implement the verify UI
- Verify OTP screen ( Includes both error and completion states when the user enters the OTP code )
- [x] Error state
- [x] Completion state

Image

### PR DESCRIPTION
**Pull request by KMTus**
## Related issues
Resolved #9 
## Summary
- Create OTP Verify screen includes: 
   + OTP imported correctly
   + OTP imported wrong 
   + Dark Mode
-  Update color
-  Update key trastion  
### step 
1. Open app/_layout.tsx 
2. Commet ``` <TamaguiProvider />```
3. Import tempalte ```import VerifyOTPTemplate from '~/components/templates/VerifyOTPTemplateTemplate'``` 
4. Call ```<VerifyOTPTemplate/>```

## Screenshots
image
image

### CODE REVIEW

**Comment by truongtdph40209** [comment]
image

Fix title translation

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]
Ảnh màn hình 2024-10-05 lúc 08 48 51

image

File: components/templates/VerifyOTPTemplate.tsx [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+} from 'react-native'
+import { Input, ScrollView, Text, View, YStack } from 'tamagui'
+
+import ContentTitle from '~/components/atoms/ContentTitle'
+import LinearGradientBackground from '~/components/molecules/LinearGradientBackground'
+import TextWithLink from '~/components/molecules/TextWithLink'
+import getColors from '~/constants/Colors'
+import useTranslation from '~/hooks/useTranslation'
+
+import DynamicIcon from '../atoms/Icons'
```
Comment (NguyenVanViet2004) [inline components/templates/VerifyOTPTemplate.tsx:18]: Relative path

File: locales/vi.json [hunk 2]
```diff
@@ -65,6 +65,14 @@
...
             "titleForgot": "Vui lòng nhập email của bạn bên dưới và chúng tôi sẽ cung cấp cho bạn mã OTP",
             "forgotPassword": "Quên mật khẩu"
+        },
+        "verify": {
+            "receiveCode": "Bạn không nhận được mã ?",
+            "resend" : "Gửi lại",
+            "titleVerify" : "Xác minh danh tính của bạn ",
+            "subVerify" : "Vui lòng nhập OTP tới 071****05. Điền xuống bên dưới.",
+            "continue" : "Tiếp tục",
+            "notification" : "Mã OTP không chính xác vui lòng thử lại!"
```
Comment (NguyenVanViet2004) [inline locales/vi.json:75]: I think it should be 'incorrectOtp'

File: locales/en.json [hunk 3]
```diff
@@ -65,6 +65,14 @@
             "sendCode": "Send code",
             "titleForgot": "Please type your email below and we will give you a OTP code",
             "forgotPassword": "Forgot your password"
+        },
+        "verify": {
+            "receiveCode": "Didn’t receive a code?",
+            "resend" : "Resend",
+            "titleVerify" : "Verify Your Identity",
+            "subVerify" : "We’ve sent a 4-digit code to 071****05. Please enter it below.",
```
Comment (NguyenVanViet2004) [inline locales/en.json:73]: `071****05` ?

File: components/atoms/Icons.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import React from 'react'
+import { Pressable, StyleSheet } from 'react-native'
+
+interface DynamicIconProps extends
+  React.ComponentProps<typeof Icons.ChevronLeft> {
+  name: keyof typeof Icons
+  onPress?: () => void
+}
+
+const DynamicIcon: React.FC<DynamicIconProps> =
```
Comment (NguyenVanViet2004) [inline components/atoms/Icons.tsx:11]: Is this component necessary?

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]

**Review by truongtdph40209 — DISMISSED** [review]

**Comment by truongtdph40209** [comment]
image

Fix dark mode backbuttton

File: app.json [hunk 5]
```diff
@@ -32,7 +32,7 @@
     },
     "orientation": "portrait",
     "icon": "./assets/images/icon.png",
-    "userInterfaceStyle": "light",
+    "userInterfaceStyle": "automatic",
```
Comment (NguyenVanViet2004) [inline app.json:35]: If you set `userInterfaceStyle` to `automatic` here, you should remove `"userInterfaceStyle": "automatic"` from lines 46 and 50.
```json
"userInterfaceStyle": "automatic",
```

File: components/molecules/Header.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  }
+
+  return (
+    <View testID="Header">
+      <XStack
+        marginTop={30}
+        gap={5}
+        alignItems="center"
+        justifyContent="center"
+      >
```
Comment (NguyenVanViet2004) [inline components/molecules/Header.tsx:45]:
```suggestion
      justifyContent="center">
```

File: components/templates/VerifyOTPTemplate.tsx [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+                      value={digit}
+                      onChangeText={(value) => {
+                        handleInputChange(value, index)
+                      }}
+                      color={colors.text}
+                      focusStyle={{
+                        borderColor: colors.blueOTP
+                      }}
+                      borderColor={
+                        isCorrect === true
```
Comment (NguyenVanViet2004) [inline components/templates/VerifyOTPTemplate.tsx:150]: You should separate it into a const variable in this component

File: components/templates/VerifyOTPTemplate.tsx [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+      }),
+      Animated.timing(shakeAnimation, {
+        duration: 100,
+        toValue: 0,
+        useNativeDriver: true
+      })
+    ]).start()
+  }
+
+  const shakeStyle = {
```
Comment (NguyenVanViet2004) [inline components/templates/VerifyOTPTemplate.tsx:83]: Create a new file for this code block

File: components/templates/VerifyOTPTemplate.tsx [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+    } else {
+      setIsCorrect(false)
+      setMessage(t('screens.verify.incorrectOtp'))
+      startShakeAnimation()
+    }
+
+    inputRefs.current.forEach((input) => input?.blur())
+  }
+
+  const startShakeAnimation = (): void => {
```
Comment (NguyenVanViet2004) [inline components/templates/VerifyOTPTemplate.tsx:63]: Create a new file for this code block

File: components/templates/VerifyOTPTemplate.tsx [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+const VerifyOTPTemplate: React.FC = (): JSX.Element => {
+  const [code, setCode] = useState<string[]>(['', '', '', ''])
+  const [message, setMessage] = useState<string>('')
+  const [shakeAnimation] = useState(new Animated.Value(0))
+  const [isCorrect, setIsCorrect] = useState<boolean | null>(null)
+  const colors = getColors(useColorScheme())
+  const inputRefs = useRef<Array<TextInput | null>>([])
+  const { t } = useTranslation()
+  const [phoneNumber] = useState<string>('0123456789')
+  useEffect(() => {
```
Comment (NguyenVanViet2004) [inline components/templates/VerifyOTPTemplate.tsx:28]: space
```suggestion

  useEffect(() => {
```

File: components/templates/VerifyOTPTemplate.tsx [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+  useColorScheme
+} from 'react-native'
+import { Input, ScrollView, Text, View, YStack } from 'tamagui'
+
+import ContentTitle from '~/components/atoms/ContentTitle'
+import Header from '~/components/molecules/Header'
+import LinearGradientBackground from '~/components/molecules/LinearGradientBackground'
+import TextWithLink from '~/components/molecules/TextWithLink'
+import getColors from '~/constants/Colors'
+import useTranslation from '~/hooks/useTranslation'
```
Comment (NguyenVanViet2004) [inline components/templates/VerifyOTPTemplate.tsx:18]: space
```suggestion
import useTranslation from '~/hooks/useTranslation'

```

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]

File: app.json (hunk 5 above)
Comment (KMTus) [reply app.json:35]: On this Android device, the minimum mode does not work, this is better

**Review by NguyenVanViet2004 — APPROVED** [review]

**Pull request merged**